python/ray/data/_internal/plan.py

```diff
@@ -503,6 +511,7 @@ def execute(
                     self,
                     allow_clear_input_blocks=allow_clear_input_blocks,
                     dataset_uuid=self._dataset_uuid,
+                    optimizer_enabled=context.optimizer_enabled,
```

Comment:
We probably can grab this out of the context instead of plumbing it through as a arg var.